python/lsst/analysis/tools/interfaces/_actions.py

```diff
@@ -251,14 +259,24 @@ def getInputSchema(self) -> KeyedDataSchema:
         yield from self.metric.getInputSchema()
         yield from self.plot.getInputSchema()
 
-    def getOutputNames(self) -> Iterable[str]:
+    def getOutputNames(self, config: pexConfig.Config | None = None) -> Iterable[str]:
         """Returns a list of names that will be used as keys if this action's
         call method returns a mapping. Otherwise return an empty Iterable
 
+        Parameters
+        ----------
+        config : `lsst.pex.config.Config`, optional
+            Configuration of the task. This is only used if the output naming
+            needs to be config-aware.
+
         Returns
         -------
         result : `Iterable` of `str`
             If a `PlotAction` produces more than one plot, this should be the
             keys the action will use in the returned `Mapping`.
         """
-        return self.plot.getOutputNames()
+        # Check if the method accepts a 'config' parameter.
```

Comment:
No, I don't like calling `inspect` here. It's not good to use this in production, and is what is called as "code smell". The `dynamicOutputNames` should tell you whether it accepts `config` or not already.

Reply:
The other way is something like:

```python
try:
    names = self.plot.getOutputNames(config=config)
except AttributeError:
    names = self.plot.getOutputNames()
return names
```
`inspect` can be very slow.

Reply:
Thanks for the suggestions! I've ditched the `inspect` approach and made the necessary tweaks using `config` itself. While the `try...except` method works, using the value of `config` seems like a more intuitive way:

``` python
if config is None:
    # `dynamicOutputNames` is set to False. 
    names = self.plot.getOutputNames()
else:
    # `dynamicOutputNames` is set to True.
    names = self.plot.getOutputNames(config=config)
return names
```

Reply:
I like checking if `config` is None as much as the try-except solution that Tim posted.

Reply:
If you know that a defined `config` means that `config=config` will work then the `if` block is fine with me. Another way to write it is:

```python
kwargs = {} if config is None else {"config": config}
return self.plot.getOutputNames(**kwargs)
```

Reply:
Yup. And just to clarify my earlier comment to Erfan, I'd like the function signature to clearly have `config` as a parameter. Passing it in as `**kwargs` is okay!

Reply:
It would be simpler if `getOutputNames` always allowed `config=None`.